### ISSUE TITLE
Is showing the skill's ID instead of its name reasonable in mattack_actors.cpp?

### ISSUE THREAD

**Issue by BrettDong** [CleverRaven/Cataclysm-DDA#20158]
When doing localization on this in #20155 , I found that in [mattack_actors.cpp:477](https://github.com/CleverRaven/Cataclysm-DDA/blob/d22e4ef2e444136302ce22a7bc7a7718bbec148b/src/mattack_actors.cpp#L477)
```C++
add_msg( m_warning, _( description.c_str() ), z.name().c_str(), _( tmp.weapon.gun_type().c_str() ) );
```
it gives a message like _A turret fires its smg!_. However, `item::gun_type()` [returns the **ID** of the skill](https://github.com/CleverRaven/Cataclysm-DDA/blob/d22e4ef2e444136302ce22a7bc7a7718bbec148b/src/item.cpp#3868):
```C++
skill_id item::gun_skill() const
{
    if( !is_gun() ) {
        return NULL_ID;
    }
    return type->gun->skill_used;
}

std::string item::gun_type() const
{
    static skill_id skill_archery( "archery" );

    if( !is_gun() ) {
        return std::string();
    }
    if( gun_skill() == skill_archery ) {
        if( ammo_type() == ammotype( "bolt" ) || typeId() == "bullet_crossbow" ) {
            return "crossbow";
        } else{
            return "bow";
        }
    }
    return gun_skill().c_str();
}
```

So it produces this:
![mattack](https://cloud.githubusercontent.com/assets/21075502/22404606/654d9bc8-e66f-11e6-989c-31fa671ac324.png)

Because 'smg' is the ident of the skill, the name of the said skill is 'submachine gun', [defined in `data/json/skills.json`](https://github.com/CleverRaven/Cataclysm-DDA/blob/d22e4ef2e444136302ce22a7bc7a7718bbec148b/data/json/skills.json#151):
```JSON
    },{
        "type" : "skill",
        "ident" : "smg",
        "name" : "submachine guns",
        "description" : "Comprised of an automatic rifle carbine designed to fire a pistol cartridge, submachine guns can reload and fire quickly, sometimes in bursts, but they are relatively inaccurate and may be prone to mechanical failures.",
        "tags" : ["combat_skill"]
    },
```

The names of the skills are localized, but their ident are not and shouldn't IMHO. Idents should be only used internally in the program, they shouldn't appear to the player.

**Comment by keyspace** [comment]
Comes from 08f2a1c4e0f210b636dfbc1486217644ae5bad34 (part of PR #15778). Before that, it was `tmp.weapon.tname().c_str()`.

**Comment by BrettDong** [comment]
Wow, thank you for providing blame info so fast!

**Comment by keyspace** [comment]
Luckily, it wasn't covered up by astyling or the like, so a depth of 0.

**Comment by keyspace** [comment]
At the time of that PR, `item::gun_compatible()` had [this](https://github.com/mugling/Cataclysm-DDA/blob/f16934056eb78bad96ba861d4fc782a3a6ee9c3d/src/item.cpp#L4215):

``` c++
} else if( !mod.type->gunmod->usable.count( gun_type() ) ) {
    msg = string_format( _( "That %s cannot be attached to a %s" ), mod.tname().c_str(), _( gun_type().c_str() ) );
```

It's been covered up by now. Not sure if that's still possible. A `g -R 'gun_type()' src` doesn't show it being localised anywhere, but may be worth a try.

**Comment by keyspace** [comment]
As to the issue question - I agree, displaying `id` instead of `name`is plain wrong. BTW, it doesn't show the exclamation mark for me - language is "system default", resulting in English.

It's fortunate there's no "crossbow turret". Imagine: _A turret fires its archery_

Or maybe _un_fortunate - would have been reported sooner.

**Comment by OzoneH3** [comment]
So, should this be changed back to ```tname()``` to show the name of the gun or
the name of the skill (which is always plural and would look really weird)?

**Comment by OzoneH3** [comment]
Fixed in #20158
